src/__tests__/surveys.js

```diff
@@ -202,6 +248,35 @@ describe('surveys', () => {
                 expect(data).toEqual([surveyWithUrl])
             })
             window.location = originalWindowLocation
+
+            // eslint-disable-next-line compat/compat
+            window.location = new URL('https://example.com/test')
+            given.surveys.getActiveMatchingSurveys((data) => {
+                expect(data).toEqual([surveyWithRegexUrl])
+            })
+            window.location = originalWindowLocation
+
+            // eslint-disable-next-line compat/compat
+            window.location = new URL('https://example.com?name=something')
```

Comment:
nice!

With the regex and wildcard options, can we move this into a separate test that specifically deals with URL matching so that this one `it` line isn't handling too many things now? 🤠

Reply:
Yes definitely!

Reply:
@liyiy should the url matching function be moved to a util? Would it be useful in other areas for comparing a url condition with the current href?